src/services/AxiosInstance.tsx

```diff
@@ -0,0 +1,53 @@
+import axios, { AxiosInstance, AxiosError, AxiosResponse } from "axios";
+import { useSelector } from "react-redux";
+
+
+export const BASE_URL = "https://e-commerce-bitcrafters-bn-1mpf.onrender.com/"
+const axiosClient = (token: string | null = null): AxiosInstance => {
```

Comment:
```suggestion
const useAxiosClient = (token?:string): AxiosInstance => {
```
Since this is is using the another hook useSelector it become also a hook, the naming of the hooks should start with `use....`.
and we can move into its own folder for `src/hooks`